app/src/main/java/org/shadowice/flocke/andotp/Dialogs/PasswordEntryDialog.java

```diff
@@ -81,10 +86,15 @@ public PasswordEntryDialog(Context context, Mode newMode, boolean blockAccessibi
 
     // TextWatcher
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (TextUtils.equals(passwordInput.getEditableText(), passwordConfirm.getEditableText()))
-            okButton.setEnabled(true);
-        else
+        if (passwordInput.getEditableText().length() >= Constants.AUTH_MIN_PASSWORD_LENGTH) {
+            toShortWarning.setVisibility(View.GONE);
+            if (TextUtils.equals(passwordInput.getEditableText(), passwordConfirm.getEditableText()))
+                okButton.setEnabled(true);
```

Comment:
You need to add an else here to disable the `okButton` again if the passwords don't match.
Otherwise it stays enabled even if you change something after they matched once.

Reply:
Fixed it in latest commit